examples/frontend/pages/share/explore/[id].tsx

```diff
@@ -0,0 +1,115 @@
+// @ts-nocheck
+/** @jsxImportSource theme-ui */
+
+import { Box, Divider } from 'theme-ui'
+import { utils } from 'near-api-js'
+import { useRouter } from 'next/router'
+import Layout from '../../../containers/Layout'
+import {
+    CreatorShare,
+    Metadata,
+    MediaObject,
+    BidCreate,
+} from '@cura/components'
+import { alertMessageState, indexLoaderState } from '../../../state/recoil'
+import { useSetRecoilState } from 'recoil'
+import {
+    useNFTContract,
+    useNFTMethod,
+    useNearHooksContainer,
+} from '@cura/hooks'
+import { getFrameWidth } from 'utils/frame-width'
+import { useStatusUpdate } from 'utils/hooks-helpers'
+import { mapPathToProject } from 'utils/path-to-project'
+
+const MARKET_SET_BID_GAS = utils.format.parseNearAmount('0.00000000020') // 200 Tgas
+
+const HARDCODED_ROYALTY_ADDRESS = process.env.YSN_ADDRESS
+const HARDCODED_ROYALTY_SHARE = '2500'
+
+const ExploreToken = () => {
+    const router = useRouter()
+    const { updateStatus } = useStatusUpdate()
+    const setAlertMessage = useSetRecoilState(alertMessageState)
+    const setIndexLoader = useSetRecoilState(indexLoaderState)
+
+    const contractAdress = router && mapPathToProject(router.asPath)
+    const project = 'share'
```

Comment:
Why is this necessary now? And was not in the deleted `share/explore.tsx` file.

Reply:
It wasn't in `share/explore.tsx` because the hooks have the default prop set to share's contract adress. And the same for Layout and project title.

And I have 2 reasons for changing that:
- I don't think it's a good idea to let the default value decide what content to get and use. Because in the future, if someone change that this project will change unexpectedly. 
I think in most case having and using a default value for anything is only good if this values are used for the majority of use cases or just as a fallback.
- Other projects gets the contract adress from `examples/frontend/utils/path-to-project.js`, so I tried to keep it consistent.

And i tried this below and didn't work, so I wrote `share` directly
`````js
    const project = `${router.query.project}`

Reply:
Makes sense @achrafismyname! Actually, if you want to create an issue to remove that `share` default value from the hooks altogether that would be great. It's simply there cause before the only project was `SHARE` and I was lazy.

>     const project = `${router.query.project}`

I might be wrong but I think this doesn't work cause on the first render the `router` object is `undefined` and it breaks.

Reply:
Thanks for reviewing this!

> I might be wrong but I think this doesn't work cause on the first render the `router` object is `undefined` and it breaks.

I tried that but it deosn't work, because the share folder isn't inside `[project]` like the other projects